src/js/Route.js

```diff
@@ -78,18 +80,26 @@ export default class Route {
      * @return {String}
      */
     compile(params) {
-        if (!this.parameterSegments.length) return this.template;
+        const segments = this.parameterSegments;
```

Comment:
This whole change is just a performance improvement, to avoid calling the getter with an expensive regex 3 times